### ISSUE TITLE
not-callable: false positive due to conflit between "next" user-function and "collections.abc"

### ISSUE THREAD

**Issue by bansan85** [PyCQA/pylint#3970]
### Steps to reproduce

```
from collections.abc import Callable


class Class:
    def next(self):
        pass

    def __init__(self):
        self.next()
```

### Current behavior

```
self.next is not callable
```

These is no warning if I remove the line ``from collections.abc import Callable``

### Expected behavior
No warning message.

### pylint --version output

```
pylint 2.6.0
astroid 2.4.2
Python 3.9.0 (tags/v3.9.0:9cf6752, Oct  5 2020, 15:34:40) [MSC v.1927 64 bit (AMD64)]
```

**Comment by hippo91** [comment]
@bansan85 thanks for your report. I can reproduce it with the following snippet:
```
#pylint: disable=missing-module-docstring, missing-function-docstring
#pylint: disable=missing-class-docstring, too-few-public-methods
from collections.abc import Callable


class Class(Callable):
    def __call__(self, *args, **kwargs):
        pass

    def next(self):
        pass

    def __init__(self):
        self.next()
```
